src/extensions/registries/page-registry.ts

```diff
@@ -1,59 +1,91 @@
 // Extensions-api -> Custom page registration
-
-import React from "react";
+import type { PageMenuTarget } from "./page-menu-registry";
+import type React from "react";
+import path from "path";
 import { action } from "mobx";
 import { compile } from "path-to-regexp";
 import { BaseRegistry } from "./base-registry";
-import { LensExtension } from "../lens-extension"
-import type { PageMenuTarget } from "./page-menu-registry";
+import { LensExtension } from "../lens-extension";
+import logger from "../../main/logger";
 
 export interface PageRegistration {
-  id: string; // will be automatically prefixed with extension name
-  routePath?: string; // additional (suffix) route path to base extension's route: "/extension/:name"
```

Comment:
This also breaks API compability.

Reply:
Added deprecation flag for now, but since we're still in beta and this was `optional` I would suggest still to break it now.

Reply:
Deprecation is better, we probably should remove it completely before GA.